internal/pkg/gateway/registry/registry_test.go

```diff
@@ -7,18 +7,188 @@ SPDX-License-Identifier: Apache-2.0
+	require.NoError(t, err, "Failed to create registry")
+
+	err = registry.addPeer("msp1", "peer2:9051")
+	require.EqualError(t, err, "unknown msp: msp1")
+	registry.msps["msp1"] = []byte(mockCert)
+	err = registry.addPeer("msp1", "peer2:9051")
+	require.NoError(t, err, "Failed to add peer")
+}
+
+func mockDiscovery() *discoveryMocks.Support {
+	discovery := &discoveryMocks.Support{}
+
+	members := []gdiscovery.NetworkMember{
+		{
+			PKIid:    []byte("id1"),
+			Endpoint: "localhost:7051",
+		},
+		{
+			PKIid:    []byte("id2"),
+			Endpoint: "peer2:9051",
+		},
+	}
+
+	infoset := []api.PeerIdentityInfo{
+		{
+			Organization: []byte("msp1"),
+			PKIId:        []byte("id1"),
+		},
+		{
+			Organization: []byte("msp1"),
+			PKIId:        []byte("id2"),
+		},
+	}
+	discovery.PeersOfChannelReturns(members)
+	discovery.IdentityInfoReturns(infoset)
+	return discovery
+}
+
+func createMockEndorsementDescriptor(t *testing.T, plan map[string][]string) *discovery.EndorsementDescriptor {
+	quantitiesByGroup := map[string]uint32{}
+	endorsersByGroups := map[string]*discovery.Peers{}
+	for group, names := range plan {
+		quantitiesByGroup[group] = 1 // for now
+		var peers []*discovery.Peer
+		for _, name := range names {
+			peers = append(peers, createMockPeer(t, name))
+		}
+		endorsersByGroups[group] = &discovery.Peers{Peers: peers}
+	}
+	descriptor := &discovery.EndorsementDescriptor{
+		Chaincode: "my_channel",
+		Layouts: []*discovery.Layout{
+			{
+				QuantitiesByGroup: quantitiesByGroup,
+			},
+		},
+		EndorsersByGroups: endorsersByGroups,
+	}
+	return descriptor
+}
+
+func createMockPeer(t *testing.T, name string) *discovery.Peer {
+	msg := &gossip.GossipMessage{
+		Content: &gossip.GossipMessage_AliveMsg{
+			AliveMsg: &gossip.AliveMessage{
+				Membership: &gossip.Member{Endpoint: name},
+			},
+		},
+	}
+
+	msgBytes, err := proto.Marshal(msg)
+	require.NoError(t, err, "Failed to create mock peer")
+
+	return &discovery.Peer{
+		StateInfo: nil,
+		MembershipInfo: &gossip.Envelope{
+			Payload: msgBytes,
+		},
+		Identity: []byte(name),
+	}
+}
+
+const mockCert = `
+-----BEGIN CERTIFICATE-----
+MIICWDCCAf2gAwIBAgIQY+mCnmNNVdmhcv1WeY5RcDAKBggqhkjOPQQDAjB2MQsw
+CQYDVQQGEwJVUzETMBEGA1UECBMKQ2FsaWZvcm5pYTEWMBQGA1UEBxMNU2FuIEZy
+YW5jaXNjbzEZMBcGA1UEChMQb3JnMS5leGFtcGxlLmNvbTEfMB0GA1UEAxMWdGxz
+Y2Eub3JnMS5leGFtcGxlLmNvbTAeFw0yMTAyMDkxNzM5MDBaFw0zMTAyMDcxNzM5
+MDBaMHYxCzAJBgNVBAYTAlVTMRMwEQYDVQQIEwpDYWxpZm9ybmlhMRYwFAYDVQQH
+Ew1TYW4gRnJhbmNpc2NvMRkwFwYDVQQKExBvcmcxLmV4YW1wbGUuY29tMR8wHQYD
+VQQDExZ0bHNjYS5vcmcxLmV4YW1wbGUuY29tMFkwEwYHKoZIzj0CAQYIKoZIzj0D
+AQcDQgAE3SvDM33N+yqTj5fjRcAAaWj8hSLb0eeG6hxWxEcTxI5Gf1tWA/pathfN
+6BiFzqBGSsWMd/X6e4HLvenzGsxW5KNtMGswDgYDVR0PAQH/BAQDAgGmMB0GA1Ud
+JQQWMBQGCCsGAQUFBwMCBggrBgEFBQcDATAPBgNVHRMBAf8EBTADAQH/MCkGA1Ud
+DgQiBCD/jZ/5NK3W/rQ/wlwRGazJF0TrnPkDQzGVlLWGYYqsGzAKBggqhkjOPQQD
+AgNJADBGAiEA78aGl4HuMb96Yy5JGZb6oHxnGntb/wFD/pP8PVZn6oACIQDdKUu2
+lUxu8s3zN5LF5eUcH7bkDfBEH0riJ/N8tqDzTA==
+-----END CERTIFICATE-----
+`
```

Comment:
Please see the many commits added to Fabric to move away from static certificates to certs generated in test and use the `tlsgen` package to create the certs that are needed to a temporary directory that lives for the duration of the test.

Reply:
thanks - I've replaced this with tlsgen